sinks/cortex/cortex_test.go

```diff
@@ -281,10 +289,12 @@ func TestChunkedWritesRespectContextCancellation(t *testing.T) {
 	// Perform the flush to the test server
 	flushResult, err := sink.Flush(ctx, metrics)
 	assert.Error(t, err)
-	assert.Equal(t, sinks.MetricFlushResult{MetricsFlushed: 4, MetricsDropped: 3, MetricsSkipped: 0}, flushResult)
+	assert.Equal(t, sinks.MetricFlushResult{MetricsFlushed: 3, MetricsDropped: 9, MetricsSkipped: 0}, flushResult)
 
 	// we're cancelling after 2 so we should only see 2 chunks written
 	assert.Equal(t, 2, len(server.History()))
+	assert.Equal(t, 3, len(server.History()[0].data.GetTimeseries()))
+	assert.Equal(t, 3, len(server.History()[1].data.GetTimeseries()))
 }
```

Comment:
Wanted to verify expected behavior here:
There are 12 total samples in the chunked_input.json. The write_batch_size is 3:
1. We write 3 successfully.
2. We write 3 more, but context gets canceled. I think there's some ambiguity about whether the write finished or was interrupted. We consider these 3 as dropped. (Q: Are we ok with this behavior? These 3 will be considered dropped, despite them possibly having already gotten forwarded and processed.)
3. The last 6 are not written at all. I've added these under dropped. (Q: Should these be considered dropped? Or skipped?)